fairscale/nn/data_parallel/fully_sharded_data_parallel.py

```diff
@@ -1470,6 +1470,17 @@ def _free_full_params(self, params: Optional[List[Parameter]] = None) -> None:
             # the Storage to 0 to save memory.
             free_storage_(p._full_param_padded)
 
+    def shard_reconstruction_info(self, consider_nested: bool = True) -> Dict[str, list]:
+        """Get the information needed to reconstruct the model from shards offline."""
```

Comment:
@QuentinDuval, since you will be using this API, may I suggest that you add to this PR:

- example usage of this function in the doctoring
- simple test that actually reconstruct the state using this metadata?

For a shorter name, perhaps rename this function to `shard_metadata_dict` to be symmetric to `state_dict`?

Reply:
Sure, I can had some documentation (I will get this PR and try it this morning). In terms of naming, I would suggest the following: `local_metadata_dict` to mirror `local_state_dict`.